modules/json/src/parsers/convert-json-expression.js

```diff
@@ -0,0 +1,74 @@
+import {get} from '../utils/get';
+
+// expression-eval: Small jsep based expression parser that supports array and object indexing
+import expressionEval from 'expression-eval';
+
+class EvaluationError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'EvaluationError';
+  }
+}
+
+const SINGLE_IDENTIFIER_REGEX = /^\s*[A-Za-z]+\s*$/;
+
+const cachedExpressionMap = {
+  '-': object => object
+};
+
+// Calculates an accessor function from a JSON string
+// '-' : x => x
+// 'a.b.c': x => x.a.b.c
+export default function convertJSONExpression(propValue, configuration, isAccessor) {
+  // NOTE: Can be null which represents invalid function. Return null so that prop can be ommitted
+  if (propValue in cachedExpressionMap) {
+    return cachedExpressionMap[propValue];
+  }
+
+  let func;
```

Comment:
Nit: Move `let func` to line 38 (principle: declare variables as close to initialization/use as possible).